src/datasource/DataSource.ts

```diff
@@ -343,6 +344,25 @@ export class SMDataSource extends DataSourceApi<SMQuery, SMOptions> {
     });
   }
 
+  //--------------------------------------------------------------------------------
+  // ALERTS PER CHECK
+  //--------------------------------------------------------------------------------
+
+  async listAlertsForCheck(checkId?: number) {
+    if (!checkId) {
+      const emptyResponse: CheckAlertsResponse = { alerts: [] };
+      return emptyResponse;
+    }
```

Comment:
Let's kill this. We can handle this problem upstream so the request never gets fired if there isn't a check id.